crates/libcontainer/src/notify_socket.rs

```diff
@@ -93,6 +96,15 @@ impl NotifyListener {
     }
 }
 
+impl Clone for NotifyListener {
+    fn clone(&self) -> Self {
+        let fd = self.socket.as_raw_fd();
+        // This is safe because we just duplicate a valid fd.
+        let socket = unsafe { UnixListener::from_raw_fd(fd) };
+        Self { socket }
+    }
+}
```

Comment:
👍 but may I ask you to add the test to ensure the same fs?

Reply:
Added a unit test to make sure the cloned listener still functions correctly.